react/week2/todolist/src/App.js

```diff
@@ -0,0 +1,65 @@
+import React, { Component } from "react";
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: [
+        {
+          id: 1,
+          description: "Get out of bed",
+        },
+        {
+          id: 2,
+          description: "Brush teeth",
+        },
+        {
+          id: 3,
+          description: "Eat breakfast",
+        },
+      ],
+    };
+  }
+
+  render() {
+    return (
+      <div>
+        <h1>todo App</h1>
+        <button onClick={this.addTodoItem}>Add Todo</button>
+        <ul>
+          {this.state.todos.map((task, i) => {
+            return (
+              <li key={i}>
+                {task.description}{" "}
+                <button data-index={i} onClick={this.deleteTodoItem}>
+                  Completed
+                </button>
+              </li>
+            );
+          })}
+        </ul>
+      </div>
+    );
+  }
+
+  addTodoItem = () => {
+    const updatedTodo = this.state.todos;
+    updatedTodo.push({
+      id: Math.floor(Math.random() * 100),
+      description:
+        updatedTodo[Math.floor(Math.random() * updatedTodo.length)].description,
+    });
+    this.setState({ todos: updatedTodo });
+  };
+
+  deleteTodoItem = (event) => {
+    const index = event.target.dataset.index;
+    this.setState((state) => {
+      const todoWithDeleted = [...state.todos];
+      todoWithDeleted.splice(index, 1);
+      return { todos: todoWithDeleted };
+    });
+  };
+}
```

Comment:
Good that you found a way to target the todo that you wount to delete. 
However you shouldn't relay on `event.target.dataset.index` to get the index of the todo that you want to delete. Instead, you can pass the `id` of todo the function and then remove the todo associated with that `id`, fx:
```suggestion
    deleteTodoItem = (id) => {
        const newTodos = this.state.todos.filter(todo => {
            return todo.id !== id;
        });
        const newTodoList = this.setState({
            todos: newTodos,
        });
    };
}
```
This would be much cleaner code and you don't have to worry and indexing.

Reply:
Great,.I have commited the changes,.,,.